### PR TITLE
feat: empty-block

### PR DESCRIPTION
**Pull request by grigory51**
Example:

```
class VxLan(PartialGenerator):
    def acl_huawei(self, device):
          return r"""
              evpn
                  mac-duplication
          """

    def run_huawei(self, device):
        with self.block("evpn"):
            with self.block("mac-duplication"):
                pass
```

As is:
```
evpn
  mac-duplication
  quit
```

To be:
```
evpn
  mac-duplication
    quit
  quit
```

### CODE REVIEW

File: annet/generators/base.py [hunk 1]
```diff
@@ -59,6 +61,8 @@ def supports_device(self, device) -> bool:  # pylint: disable=unused-argument
 
 
 class TreeGenerator(BaseGenerator):
+    block_holder = BLOCK_HOLDER
```
Comment (azryve) [inline annet/generators/base.py:64]: Вроде нигде не используется?

Comment (azryve) [reply annet/generators/base.py:64]: А, понял

**Comment by azryve** [comment]
Is there any particular reason not to make this behaviour for the empty blocks such as
```  
with self.block("mac-duplication"):
    pass
```
I would argue that it's a more natural way. If it's done on purpose, could you please add an explicit test for these cases in test_generators.

File: annet/types.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
 from collections import OrderedDict
 from typing import Any, Dict, List, MutableMapping, NamedTuple, Optional, Tuple, Union
-
+from uuid import uuid4
 from annet.annlib.types import Op  # pylint: disable=unused-import
 
 from annet.storage import Device, Storage
 
+BLOCK_HOLDER = f"empty-block-{uuid4().hex}"
```
Comment (gescheit) [inline annet/types.py:8]: Random data seems unnecessary here

**Pull request closed without merging**